core/trino-main/src/main/java/io/trino/sql/planner/PartitioningScheme.java

```diff
@@ -77,6 +81,7 @@ public PartitioningScheme(
         checkArgument(!replicateNullsAndAny || columns.size() <= 1, "Must have at most one partitioning column when nullPartition is REPLICATE.");
         this.replicateNullsAndAny = replicateNullsAndAny;
         this.bucketToPartition = requireNonNull(bucketToPartition, "bucketToPartition is null");
+        this.partitionCount = requireNonNull(partitionCount, "partitionCount is null");
```

Comment:
For now please add a check that either `partitionCount` is empty or partitioning is `SystemPartitioningHandle`